src/main/java/com/commercetools/project/sync/customobject/CustomObjectSyncer.java

```diff
@@ -0,0 +1,135 @@
+        CustomObjectDraft<JsonNode>,
+        CustomObjectSyncStatistics,
+        CustomObjectSyncOptions,
+        CustomObjectQuery<JsonNode>,
+        CustomObjectSync> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CustomObjectSyncer.class);
+
+  /**
+   * Instantiates a {@link Syncer} which is used to sync resources from a source to a target
+   * commercetools project.
+   *
+   * @param sync The sync module that is used for syncing the resource drafts to the target project,
+   *     after being transformed from the resources fetched from the source project.
+   * @param sourceClient the client used for querying data from the source commercetools project.
+   * @param targetClient the client used for syncing the transformed drafts into the target
+   *     commercetools project.
+   * @param customObjectService service that is used for fetching and persisting the last sync
+   *     timestamp for delta syncing.
+   * @param clock the clock to record the time for calculating the sync duration.
+   */
+  private CustomObjectSyncer(
+      @Nonnull CustomObjectSync sync,
+      @Nonnull SphereClient sourceClient,
+      @Nonnull SphereClient targetClient,
+      @Nonnull CustomObjectService customObjectService,
+      @Nonnull Clock clock) {
+    super(sync, sourceClient, targetClient, customObjectService, clock);
+  }
+
+  @Nonnull
+  @Override
+  protected CompletionStage<List<CustomObjectDraft<JsonNode>>> transform(
+      @Nonnull List<CustomObject<JsonNode>> page) {
+    final List<CustomObjectDraft<JsonNode>> collect =
+        page.stream()
+            .map(
+                customObject ->
+                    CustomObjectDraft.ofUnversionedUpsert(
+                        customObject.getContainer(),
+                        customObject.getKey(),
+                        customObject.getValue()))
+            .collect(Collectors.toList());
+    return CompletableFuture.completedFuture(collect);
+  }
+
+  @Nonnull
+  @Override
+  protected CustomObjectQuery<JsonNode> getQuery() {
+    // todo: exclude custom objects created by the project-sync from syncing
+    return CustomObjectQuery.ofJsonNode();
+  }
+
+  @Nonnull
+  public static CustomObjectSyncer of(
+      @Nonnull final SphereClient sourceClient,
+      @Nonnull final SphereClient targetClient,
+      @Nonnull final Clock clock) {
+    final QuadConsumer<
+            SyncException,
+            Optional<CustomObjectDraft<JsonNode>>,
+            Optional<CustomObject<JsonNode>>,
+            List<UpdateAction<CustomObject<JsonNode>>>>
+        logErrorCallback =
+            (exception, newResourceDraft, oldResource, updateActions) -> {
+              final String resourceIdentifier = getCustomObjectResourceIdentifier(oldResource);
+              updateActions = updateActions == null ? Collections.emptyList() : updateActions;
+              logErrorCallback(
+                  LOGGER, "customObject", exception, resourceIdentifier, updateActions);
```

Comment:
We know there should have no update actions in CO. Should we explicitly pass Collections.emptyList() here to override what caller given?